ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -473,8 +471,7 @@ func (r *RayClusterReconciler) reconcileRouteOpenShift(ctx context.Context, inst
 func (r *RayClusterReconciler) reconcileIngressKubernetes(ctx context.Context, instance *rayv1.RayCluster) error {
 	logger := ctrl.LoggerFrom(ctx)
 	headIngresses := networkingv1.IngressList{}
-	filterLabels := client.MatchingLabels{utils.RayClusterLabelKey: instance.Name}
-	if err := r.List(ctx, &headIngresses, client.InNamespace(instance.Namespace), filterLabels); err != nil {
+	if err := r.List(ctx, &headIngresses, common.RayClusterAllPodsAssociationOptions(instance).ToListOptions()...); err != nil {
```

Comment:
This is pretty confusing. We use `RayClusterAllPodsAssociationOptions` to list ingress.